mne/viz/_3d.py

```diff
@@ -335,98 +367,100 @@ def plot_trans(info, trans='auto', subject=None, subjects_dir=None,
     elif not isinstance(trans, dict):
         raise TypeError('trans must be str, dict, or None')
     head_mri_t = _ensure_trans(trans, 'head', 'mri')
+    dev_head_t = info['dev_head_t']
     del trans
 
+    # Figure out our transformations
+    if coord_frame == 'meg':
+        head_trans = invert_transform(dev_head_t)
+        meg_trans = Transform('meg', 'meg')
+        mri_trans = combine_transforms(dev_head_t, head_mri_t, 'mri', 'meg')
+    elif coord_frame == 'mri':
+        head_trans = head_mri_t
+        meg_trans = combine_transforms(dev_head_t, head_mri_t, 'meg', 'mri')
+        mri_trans = Transform('mri', 'mri', np.eye(4))
+    else:  # coord_frame == 'head'
+        head_trans = Transform('head', 'head', np.eye(4))
+        meg_trans = info['dev_head_t']
+        mri_trans = invert_transform(head_mri_t)
+
     # both the head and helmet will be in MRI coordinates after this
     meg_picks = pick_types(info, meg=True, ref_meg=ref_meg)
     surfs = dict()
     if show_head:
         subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
         surfs['head'] = get_head_surf(subject, source=source,
                                       subjects_dir=subjects_dir)
-    if (ch_type is None and len(meg_picks) > 0) or ch_type == 'meg':
+    if 'helmet' in meg_sensors and len(meg_picks) > 0 and \
+            (ch_type is None or ch_type == 'meg'):
         surfs['helmet'] = get_meg_helmet_surf(info, head_mri_t)
-    if coord_frame == 'meg':
-        surf_trans = combine_transforms(info['dev_head_t'], head_mri_t,
-                                        'meg', 'mri')
-    elif coord_frame == 'head':
-        surf_trans = head_mri_t
-    else:  # coord_frame == 'mri'
-        surf_trans = None
     for key in surfs.keys():
-        surfs[key] = transform_surface_to(surfs[key], coord_frame, surf_trans)
-    del surf_trans
+        surfs[key] = transform_surface_to(surfs[key], coord_frame, mri_trans)
 
     # determine points
     meg_rrs, meg_tris = list(), list()
     hpi_loc = list()
     ext_loc = list()
     car_loc = list()
     eeg_loc = list()
-    if eeg_sensors and (ch_type is None or ch_type == 'eeg'):
+    eegp_loc = list()
+    if len(eeg_sensors) > 0 and (ch_type is None or ch_type == 'eeg'):
         eeg_loc = np.array([info['chs'][k]['loc'][:3]
-                           for k in pick_types(info, meg=False, eeg=True)])
+                           for k in pick_types(info, meg=False, eeg=True,
+                                               ref_meg=False)])
         if len(eeg_loc) > 0:
-            # Transform EEG electrodes from head coordinates if necessary
-            if coord_frame == 'meg':
-                eeg_loc = apply_trans(invert_transform(info['dev_head_t']),
-                                      eeg_loc)
-            elif coord_frame == 'mri':
-                eeg_loc = apply_trans(head_mri_t, eeg_loc)
+            eeg_loc = apply_trans(head_trans, eeg_loc)
```

Comment:
no, `eeg_loc` needs to get transformed to the proper coordinate frame regardless of whether or not `eegp_loc` will be used